app/models/group.rb

```diff
@@ -213,19 +213,7 @@ def default_community
   def default_user(uid)
     user = User.find_by(uid:)
     return user if user.present?
-
-    begin
-      user = User.new(uid:, default_group_id: id)
-      user.save!
-      user
-    rescue ActiveRecord::RecordNotUnique => unique_error
-      # If adding a submitter to an existing group
-      Rails.logger.error("Failed to created a new user for #{self}: #{unique_error}")
-      user = User.new(uid:)
-      user.default_group_id = id
-      user.save!
-      user
-    end
+    User.new_for_uid(uid, id)
```

Comment:
Our existing `new_for_id` method did most of what we needed, I just updated it to accept a default group id and that fixes the problem.